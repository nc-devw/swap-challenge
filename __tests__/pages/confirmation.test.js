import { render } from "@testing-library/react";
import Confirmation from "@/pages/confirmation";

let transaction = {
  input: { iconUrl: "http://test", quantity: 0 },
  output: { iconUrl: "http://test", quantity: 0 },
};
const mockPush = jest.fn();

jest.mock("next/router", () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

jest.mock("@/context/global", () => ({
  useAppContext: () => ({
    transaction: transaction,
  }),
}));

describe("Confirmation", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Confirmation />);
    expect(asFragment()).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should redirect to swap", () => {
    transaction.input.iconUrl = "";
    render(<Confirmation />);
    expect(mockPush).toBeCalledTimes(1);
    expect(mockPush).toBeCalledWith("/swap");
  });
});
