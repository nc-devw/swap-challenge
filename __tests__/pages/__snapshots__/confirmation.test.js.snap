// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation should render correctly 1`] = `[Function]`;
